examples/demo.rs

```diff
@@ -52,8 +52,13 @@ fn main() -> std::result::Result<(), Box<dyn Error>> {
     let property_id = env::var("PROPERTY_ID").expect("PROPERTY_ID should be set.");
 
     let configuration = ConfigurationId::new(guid, environment_id, collection_id);
-    let client =
-        AppConfigurationClientIBMCloud::new(&apikey, &region, configuration, OfflineMode::Fail)?;
+    let client = AppConfigurationClientIBMCloud::new(
+        &apikey,
+        &region,
+        configuration,
+        OfflineMode::Fail,
+        false,
```

Comment:
Guess this is OK for now, but adding a boolean flag here might cause a bit confusion on the user-side.
It is bloating up construction with a rarely used constructor argument.

Should we provide two constructors instead?
`new()` and `new_using_private_cloud()`?